apps/vs-code-designer/src/app/commands/workflows/unitTest/saveBlankUnitTest.ts

```diff
@@ -2,29 +2,491 @@
+    parsedOutputs.outputParameters[parameterKey] = {
+      outputs: transformRawOutputs(unitTestDefinition['outputParameters'][parameterKey].outputs),
+    };
+  }
+
+  return parsedOutputs;
+}
+
+/**
+ * Generates a codeful unit test by calling the backend API, unzipping the response, and creating the .cs file.
+ * @param {IAzureConnectorsContext} context - The context for Azure Connectors.
+ * @param {string} projectPath - The path to the project directory.
+ * @param {string} workflowName - The name of the workflow for which the test is being created.
+ * @param {string} unitTestName - The name of the unit test to be created.
+ * @param {string | undefined} runId - The ID of the run.
+ * @returns {Promise<void>} - A promise that resolves when the unit test has been generated.
+ */
+async function generateBlankCodefulUnitTest(
+  context: IAzureConnectorsContext,
+  projectPath: string,
+  workflowName: string,
+  unitTestName: string
+): Promise<void> {
   try {
-    const workflowName = ''; // TODO: Determine workflow name dynamically
-    const unitTestName = ''; // TODO: Prompt user for unit test name or derive programmatically
+    // Destructure testsDirectory in addition to other paths
+    const { testsDirectory, logicAppName, logicAppFolderPath, workflowFolderPath, unitTestFolderPath } = getUnitTestPaths(
+      projectPath,
+      workflowName,
+      unitTestName
+    );
+    // Ensure the required directories exist
+    await fs.ensureDir(logicAppFolderPath);
+    await fs.ensureDir(workflowFolderPath);
+    await fs.ensureDir(unitTestFolderPath!);
+    const csprojFilePath = path.join(logicAppFolderPath, `${logicAppName}.csproj`);
+    await createCsFile(unitTestFolderPath!, unitTestName, workflowName);
+    // Generate the .csproj file if it doesn't exist
+    if (!(await fs.pathExists(csprojFilePath))) {
+      ext.outputChannel.appendLog(localize('creatingCsproj', 'Creating .csproj file at: {0}', csprojFilePath));
+      await createCsprojFile(csprojFilePath, logicAppName);
+    }
+    vscode.window.showInformationMessage(
+      localize('info.generateCodefulUnitTest', 'Generated unit test "{0}" in "{1}"', unitTestName, unitTestFolderPath)
+    );
+    // Check if testsDirectory is already part of the workspace
+    const workspaceFolders = vscode.workspace.workspaceFolders || [];
+    const isTestsDirectoryInWorkspace = workspaceFolders.some((folder) => folder.uri.fsPath === testsDirectory);
+    if (!isTestsDirectoryInWorkspace) {
+      // Add testsDirectory to workspace if not already included
+      ext.outputChannel.appendLog(localize('addingTestsDirectory', 'Adding tests directory to workspace: {0}', testsDirectory));
+      FileManagement.addFolderToWorkspace(testsDirectory);
+    }
+    context.telemetry.properties.unitTestGenerationStatus = 'Success';
+  } catch (error: any) {
+    context.telemetry.properties.unitTestGenerationStatus = 'Failed';
+    if (error.code) {
```

Comment:
NIT: can we log the error message as well? Is the scenario in which there is not error.code useful for telemetry?
To get the message you can try this
``` typescript
const message = error instanceof Error ? error.message : typeof error === 'string' ? error : 'Unknown error';
```